service/history/historyEngine.go

```diff
@@ -316,9 +270,12 @@ func (e *historyEngineImpl) Stop() {
 		e.visibilityProcessor.Stop()
 	}
 
+	e.clusterMetadata.UnRegisterMetadataChangeCallback(fmt.Sprintf(historyServiceCallbackID, e.shard.GetShardID()))
```

Comment:
I think it is worth to define a private method as getMetadataChangeCallbackID(), then you don't need to define historyServiceCallbackID.
```
func getMetadataChangeCallbackID() string {
  return fmt.Sprintf("history-service-%d", e.shard.GetShardID())
}
```

Reply:
done